infinispan-embedded/src/main/java/org/hibernate/ogm/datastore/infinispan/InfinispanDialect.java

```diff
@@ -192,6 +195,14 @@ public boolean isStoredInEntityStructure(AssociationKeyMetadata associationKeyMe
 	@Override
 	//TODO should we use GridTypes here?
 	public Number nextValue(NextValueRequest request) {
+
+		// if transport is defined on Infinispan cache manager
+		// we can use clustered counter to generate next value
+		EmbeddedCacheManager cacheManager = getCacheManager().getCacheManager();
```

Comment:
What happens if the transport is not defined? Is there an exception?
If not, would it make sense to add one? This way we can tell the user that something is missing if he wants to generate values

Reply:
At the moment I left the fallback to default behavior. But as you wish, in fact it could make sense to make the transport mandatory in case of sequence use. Do we throw an exception? At that point we could also remove the cache of the sequences that we will not use anymore. But before doing it I wanted your feedback.

Reply:
The problem is that the previous code is not correct (I thought it was but I was proved wrong).

I was hoping that we'd now be able to use the `CounterManager` to entirely replace (and fix) the current approach.

Reply:
Also for performance reasons could you avoid extracting the `EmbeddedCacheManager` each time a `nextValue` is needed?

Reply:
absolutely